### PR TITLE
EL-1217: Dual running new UAT namespace

### PR DESCRIPTION
**Pull request by HettieS**
[Jira ticket](https://dsdmoj.atlassian.net/browse/EL-1217)

## What changed and why

This PR gets dual deployments running to old and new UAT namespaces. Alongside changes here, changes are made in Circle CI Config and `kube-secrets`. These changes are detailed on the ticket. The same approach should be able to be used for staging and prod. 

This PR also makes changes to _env.tpl and the deploy script which should set the groundwork for dual running for other environments.

We are temporarily referring the the new namespace UAT environment as `ccq-uat` so that we can deploy successfully to a different URL.

## Checklist

Before you ask people to review this PR:

- Tests and rubocop should be passing
- Branch is generally up to date with main Github - definitely no conflicts
- No unnecessary whitespace changes. These make diffs harder to read and conflicts more likely.
- PR description says *what* changed and *why*, with a link to the JIRA story.
- Diff has been checked for unexpected changes being included.
- Commit messages say why the change was made.

### CODE REVIEW

**Comment by HettieS** [comment]
I ran the redis script we wrote in https://github.com/ministryofjustice/laa-estimate-financial-eligibility-for-legal-aid/pull/1012 as a way of testing the import of data across UATs. 

There were some small issues importing old namespace UAT redis into new namespace UAT redis, I tried this three times and each time, the migration failed due to 1 or 2 'missing' keys. But we think this is because new keys were being generated as the migration was running.

The verify this, we amended the script slightly so that we could import new namespace UAT redis into new namespace UAT staging (we knew no one was using these at the time). The script imported successfully with no failures. This implies the first few fails were due to new keys being created at time of migration.

As we will put the site in maintenance mode when we run this in production, we think that should eliminate this type of failure, and it should run successfully.

File: helm_deploy/laa-estimate-eligibility/values/ccq-uat.yaml
Comment (patrick-laa) [inline helm_deploy/laa-estimate-eligibility/values/ccq-uat.yaml:44]: My assumption is that we can't use the actual main UAT URL as the host name here because it's already in use in the old namespace and something will break if we declare the same host for both. But did we actually test that? If the weight is set to 0 here, would it actually _allow_ us to have both namespaces declare the same host?

I'm asking because we need to solve this problem at some point, and I've got solving this problem earmarked for EL-1214 which on JIRA is marked as blocking this ticket. If you've learned anything about this as part of this ticket, I'm eager to see if we can use it to make progress on EL-1214.

File: helm_deploy/laa-estimate-eligibility/values/ccq-uat.yaml
Comment (patrick-laa) [inline helm_deploy/laa-estimate-eligibility/values/ccq-uat.yaml:109]: This should probably be set to `main-ccq-uat.cloud-platform.service.justice.gov.uk`, since the other "main" host might go away with the other namespace. But if so, we need to whitelist `main-ccq-uat...` in the Google Developer Console as a valid redirect URI for OAuth.

File: bin/deploy
Comment (patrick-laa) [reply bin/deploy:10]: Ooh, could we put a BRANCH_HOST_SUFFIX in the two context we have, and in the `laa-estimate-eligibility-uat` context it could be `check-client-qualifies-legal-aid-uat`, but in the `laa-check-client-qualifies-uat` one it could be `ccq-uat` maybe? THat way this line could be:

```bash
RELEASE_HOST="$BRANCH_RELEASE_NAME-$BRANCH_HOST_SUFFIX.cloud-platform.service.justice.gov.uk"
```

And we'd end up with more concise UAT URLs?

Comment (willc-work) [reply bin/deploy:10]: I am not even sure this $ENVIRONMENT is required, this script always deploys to UAT so it should never have a production or staging.

File: helm_deploy/laa-estimate-eligibility/values/ccq-uat.yaml
Comment (willc-work) [reply helm_deploy/laa-estimate-eligibility/values/ccq-uat.yaml:48]: just a fail safe we agreed on, when host is changed this will already be at zero.

File: helm_deploy/laa-estimate-eligibility/values/ccq-uat.yaml
Comment (willc-work) [reply helm_deploy/laa-estimate-eligibility/values/ccq-uat.yaml:44]: no we did not test it because testing it could only happen after merging the PR so we are playing it safe by changing it for now. Once we can dual deploy we can run a branch to test exactly how it works, as this will apply to UAT/STG and PROD envs

File: bin/deploy
Comment (HettieS) [reply bin/deploy:10]: Yes that is a nice idea! We should try and do that. I can take a look!

File: helm_deploy/laa-estimate-eligibility/values/ccq-uat.yaml
Comment (willc-work) [reply helm_deploy/laa-estimate-eligibility/values/ccq-uat.yaml:109]: we were not 100% sure of the correct approach for this, will make those changes.

File: helm_deploy/laa-estimate-eligibility/values/ccq-uat.yaml
Comment (HettieS) [reply helm_deploy/laa-estimate-eligibility/values/ccq-uat.yaml:48]: This was all quite trial and error - @willc-work and I had a good chat with @exonian about ingress settings. I think because we had managed to get the deployment working in this way, we stuck with it. I think 0 in this particular context might have the same affect as setting to 100? But you're right, we have changed the identifier, ingress and host so perhaps we don't need this. I will defer to Will and Michael!

File: helm_deploy/laa-estimate-eligibility/values/ccq-uat.yaml
Comment (HettieS) [reply helm_deploy/laa-estimate-eligibility/values/ccq-uat.yaml:109]: Happy to make this change although I'm not sure how to do the whitelisting, if someone could help with that @patrick-laa @willc-work

Comment (patrick-laa) [reply helm_deploy/laa-estimate-eligibility/values/ccq-uat.yaml:109]: Sure @HettieS I think this link will get you to the right screen: https://console.cloud.google.com/apis/credentials/oauthclient/276075615370-ncqvmdi84jruv1jggfg4bu8nlvk1drj0.apps.googleusercontent.com?orgonly=true&project=ccq-admin&supportedpurview=organizationId - all you need to do is add a new entry to the list of 'Authorised redirect URIs' and save.

Comment (HettieS) [reply helm_deploy/laa-estimate-eligibility/values/ccq-uat.yaml]: thank you! this should now be done

**Review by willc-work — APPROVED** [review]
Approving PR changes

**Pull request merged**
